server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -0,0 +1,745 @@
+                lowerTerm == null ? null : indexedValueForSearch(rewriteLowerTerm),
+                upperTerm == null ? null : indexedValueForSearch(rewriteUpperTerm),
+                includeLower,
+                includeUpper
+            );
+        }
+
+        /**
+         * if there is dot path. query the field name in flatObject parent field.
+         * else query in _field_names system field
+         */
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            String searchKey;
+            String searchField;
+            if (name().contains(DOT_SYMBOL)) {
+                searchKey = name().split("\\.")[0];
+                searchField = name();
+            } else {
+                searchKey = FieldNamesFieldMapper.NAME;
+                searchField = name();
+            }
+            return new TermQuery(new Term(searchKey, indexedValueForSearch(searchField)));
+        }
+
+        @Override
+        public Query wildcardQuery(
+            String value,
+            @Nullable MultiTermQuery.RewriteMethod method,
+            boolean caseInsensitve,
+            QueryShardContext context
+        ) {
+            // flat_object field types are always normalized, so ignore case sensitivity and force normalize the wildcard
+            // query text
+            throw new QueryShardException(
+                context,
+                "Can only use wildcard queries on keyword and text fields - not on [" + name() + "] which is of type [" + typeName() + "]"
+            );
+        }
+
+    }
+
+    private final ValueFieldMapper valueFieldMapper;
+    private final ValueAndPathFieldMapper valueAndPathFieldMapper;
+    private final IndexAnalyzers indexAnalyzers;
+
+    FlatObjectFieldMapper(
+        String simpleName,
+        FieldType fieldType,
+        FlatObjectFieldType mappedFieldType,
+        ValueFieldMapper valueFieldMapper,
+        ValueAndPathFieldMapper valueAndPathFieldMapper,
+        MultiFields multiFields,
+        CopyTo copyTo,
+        Builder builder
+    ) {
+        super(simpleName, fieldType, mappedFieldType, copyTo);
+        assert fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS) <= 0;
+        this.fieldType = fieldType;
+        this.indexAnalyzers = builder.indexAnalyzers;
+        this.valueFieldMapper = valueFieldMapper;
+        this.valueAndPathFieldMapper = valueAndPathFieldMapper;
+        this.mappedFieldType = mappedFieldType;
+    }
+
+    @Override
+    protected FlatObjectFieldMapper clone() {
+        return (FlatObjectFieldMapper) super.clone();
+    }
+
+    @Override
+    protected void mergeOptions(FieldMapper other, List<String> conflicts) {
+
+    }
+
+    @Override
+    public FlatObjectFieldType fieldType() {
+        return (FlatObjectFieldType) super.fieldType();
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context) throws IOException {
+        String value = null;
```

Comment:
Since there are two branches (if / else), I would use two variables instead of the common one `value`